backend/src/nodes/image_util_nodes.py

```diff
@@ -5,80 +5,98 @@
+from .utils.image_utils import blend_images, calculate_ssim
 from .utils.pil_utils import *
 from .utils.utils import get_h_w_c
 
+ndarray32 = npt.NDArray[np.float32]
 
-@NodeFactory.register("chainner:image:overlay")
-class ImOverlay(NodeBase):
-    """OpenCV transparency overlay node"""
+
+@NodeFactory.register("chainner:image:blend")
+class ImBlend(NodeBase):
+    """Blending mode node"""
 
     def __init__(self):
         """Constructor"""
         super().__init__()
-        self.description = "Overlay transparent images on base image."
+        self.description = """Blends overlay image onto base image using 
+            specified mode and opacities."""
         self.inputs = [
-            ImageInput("Base").with_id(0),
+            ImageInput("Base"),
             SliderInput(
                 "Base Opacity",
                 maximum=100,
                 default=100,
                 step=0.1,
                 controls_step=1,
                 unit="%",
-            ).with_id(5),
-            ImageInput("Overlay A").with_id(1),
-            SliderInput(
-                "Opacity A",
-                maximum=100,
-                default=50,
-                step=0.1,
-                controls_step=1,
-                unit="%",
-            ).with_id(2),
-            ImageInput("Overlay B").make_optional().with_id(3),
+            ),
+            ImageInput("Overlay").make_optional(),
             SliderInput(
-                "Opacity B",
+                "Overlay Opacity",
                 maximum=100,
-                default=50,
+                default=100,
                 step=0.1,
                 controls_step=1,
                 unit="%",
-            ).with_id(4),
+            ),
+            DropDownInput(
+                "Blend Mode",
+                [
+                    {"option": "Normal", "value": 0},
+                    {"option": "Multiply", "value": 1},
+                    {"option": "Darken", "value": 2},
+                    {"option": "Lighten", "value": 3},
+                    {"option": "Add", "value": 4},
+                    {"option": "Color Burn", "value": 5},
+                    {"option": "Color Dodge", "value": 6},
+                    {"option": "Reflect", "value": 7},
+                    {"option": "Glow", "value": 8},
+                    {"option": "Overlay", "value": 9},
+                    {"option": "Difference", "value": 10},
+                    {"option": "Negation", "value": 11},
+                    {"option": "Screen", "value": 12},
+                    {"option": "Xor", "value": 13},
+                ],
+            ),
         ]
         self.outputs = [ImageOutput()]
         self.category = IMAGE_UTILITY
-        self.name = "Overlay Images"
+        self.name = "Blend Images"
         self.icon = "BsLayersHalf"
         self.sub = "Compositing"
 
     def run(
         self,
-        base: np.ndarray,
+        base: ndarray32,
```

Comment:
Was it really a good idea to merge what could have been 2 PRs (better type annotations + blend node)?

Reply:
Ehh, maybe not. It only happened because I was trying to see if there was a way to get rid of a pyright type error I was getting, and then I just made it consistent across the files I was working on.

Reply:
> get rid of a pyright type error

Well, that didn't exactly work out...

Reply:
Annoyingly.